cmd/minishift/cmd/start_preflight.go

```diff
@@ -338,17 +345,44 @@ func checkLibvirtDefaultNetworkActive() bool {
 
 // checkHypervDriverSwitch returns true if Virtual Switch has been selected
 func checkHypervDriverSwitch() bool {
-	switchEnv := os.Getenv("HYPERV_VIRTUAL_SWITCH")
-	if switchEnv == "" {
+	switchEnvName := "HYPERV_VIRTUAL_SWITCH"
+	posh := powershell.New()
+	defer posh.Close()
+
+	switchEnvValue := os.Getenv(switchEnvName)
+
+	if switchEnvValue == "" {
+		checkIfDefaultSwitchExists := `Get-VMSwitch -Id c08cb7b8-9b3c-408e-8e30-5e16a3aeb444 | ForEach-Object { $_.SwitchType }`
+		fmt.Printf("\n   'Default Switch' ... ")
+		stdOut, _ := posh.Execute(checkIfDefaultSwitchExists)
+		if strings.Contains(stdOut, "Internal") {
+
+			// force setting the environment variable
+			os.Setenv("HYPERV_VIRTUAL_SWITCH", "Default Switch")
```

Comment:
on non English windows locale this switch has different names. In Polish it is 'Przełącznik domyślny'
it is possible to use this magic id all the way down to libmashine?

Reply:
I have to look into this. The switch can be identified with a UUID, so probably have to test a little with setting this.

Reply:
we can form the GUID from string:
```posh
$id=[guid]"c08cb7b8-9b3c-408e-8e30-5e16a3aeb444"
```

Reply:
I will solve this is another issue, as this got filed as #2096. WDYT?

Reply:
@gbraad Agreed we can do some more investigation as part of #2096 ... but i'd like to this change added.

Reply:
It is a problem that lies deeper inside docker/machine => https://github.com/docker/machine/issues/4429